### PR TITLE
 ✨ Append co-authors to commit message

### LINKED ISSUE #8: Implement `pre-commit` hook to append pairs
We can use existing [Git Hooks](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks) to customize `pre-commit` action.

Based in https://github.com/thechutrain/git-pair implementation, we can do something similar:

```
#!/bin/sh
set -e

# Hook from git-pair 🍐
git-pair _append-co-authors $@ #adds all of the arguments in bash
```

## Examples

-   [Pre-commit Template](https://github.com/git/git/blob/master/templates/hooks--pre-commit.sample)

### PR DESCRIPTION
**Pull request by amadeu01**
We open the file `./.git/COMMIT_EDITMSG` by using the append mode of `File.open`, then we build the collaborators list.

### CODE REVIEW

File: lib/git_pair/actions.ex [hunk 1]
```diff
@@ -52,6 +54,32 @@ defmodule GitPair.Actions do
     output(result, "Stopped pairing with everyone")
   end
 
+  def _modify_commit_msg(path) do
+    co_authors_message = IO.iodata_to_binary(make_co_authored_by)
+
+    File.open(path, [:append])
```
[the remark this answers is not in the document]
Comment (amadeu01) [reply lib/git_pair/actions.ex]: Done in https://github.com/elixirsc/git-pair/pull/30/commits/698ffbb0c34206d18e6621e4636f3178b3456e8b

**Review by wevtimoteo — APPROVED** [review]

File: lib/git_pair/actions.ex [hunk 2]
```diff
@@ -52,6 +54,32 @@ defmodule GitPair.Actions do
...
+        |> (&(List.delete_at(&1, length(&1) - 1))).()
+      _ ->
+        []
+    end
+  end
+
+  def make_co_authored_by() do
+    "\n" <> (Enum.map(collaborators, fn collaborator ->
+      "#{collaborator} <#{collaborator}@users.noreply.github.com>"
+    end) |> Enum.join("\n"))
```
[the remark this answers is not in the document]
Comment (wevtimoteo) [reply lib/git_pair/actions.ex:78]: We need to disable this rule. Will do in a separate PR.

File: lib/git_pair/actions.ex [hunk 3]
```diff
@@ -52,6 +54,32 @@ defmodule GitPair.Actions do
...
+    |> IO.binwrite(co_authors_message)
+
+    {:ok, "Success! Co-authors registered."}
+  end
+
+  defp collaborators() do
+    case command("--get-all") do
+      {collaborators, 0} ->
+        String.split(collaborators, "\n")
+        |> (&(List.delete_at(&1, length(&1) - 1))).()
```
[the remark this answers is not in the document]
Comment (wevtimoteo) [reply lib/git_pair/actions.ex:69]: We need to disable this rule. Will do in a separate PR.

File: lib/git_pair/actions.ex [hunk 4]
```diff
@@ -41,9 +44,8 @@ defmodule GitPair.Actions do
   end
 
   def status() do
-    result = command("--get-all")
-
-    output(result, "Pairing with: ")
+    "Pairing with: \n\n" <> Enum.join(collaborators, "\n")
+    |> output()
```
[the remark this answers is not in the document]
Comment (wevtimoteo) [reply lib/git_pair/actions.ex:46]: We need to disable this rule. Will do in a separate PR.

**Comment by amadeu01** [comment]
🚀  🚀 🚀 🚀 🚀 🚀 🚀 🚀

**Review by leandroh — APPROVED** [review]
🚀

**Pull request merged**